jerry-libc/jerry-libc.c

```diff
@@ -276,10 +275,14 @@ rand (void)
 void
 srand (unsigned int seed) /**< new seed */
 {
-  libc_random_gen_state[0] =
-  libc_random_gen_state[1] =
-  libc_random_gen_state[2] =
-  libc_random_gen_state[3] = seed;
+  _x = (unsigned) ((seed * 14316555781)
+       + (seed * 1183186591)
+       + (seed * 622729787)
+       + (seed * 338294347));
+
+  _y = initial_states[0];
+  _z = initial_states[1];
+  _w = initial_states[2];
```

Comment:
Here, this approach is not bad. It only needs to be adapted to the original state representation. Something like:

```c
  libc_random_gen_state[0] = (unsigned) ((seed * 14316555781)
                                         + (seed * 1183186591)
                                         + (seed * 622729787)
                                         + (seed * 338294347));

  libc_random_gen_state[1] = 842502087;
  libc_random_gen_state[2] = 3579807591;
  libc_random_gen_state[3] = 273326509;
```